src/ripple/protocol/Serializer.h

```diff
@@ -49,14 +49,12 @@ class Serializer
         mData.reserve (n);
     }
 
-    Serializer (void const* data,
-        std::size_t size)
+    Serializer (void const* data, std::size_t size)
     {
         mData.resize(size);
-        std::memcpy(mData.data(),
-            reinterpret_cast<
-                unsigned char const*>(
-                    data), size);
+
+        if (size)
+            std::memcpy(mData.data(), reinterpret_cast<unsigned char const*>(data), size);
```

Comment:
Same as the previous comment.

Reply:
I'm not sure I follow. We'll only ever enter this if `size != 0` which requires that `data != nullptr`. I guess we could add an `assert` to that effect, but, I think the better long-term option is to adopt `std::fill_n` here.

Reply:
I think that assert ((size == 0) == (data == nullptr)) is reasonable pending the larger refactor.

Reply:
Based on the potential of the caller passing `data` as null and `size` as greater than zero. The assert is fine.